tests/test_Dataset.py

```diff
@@ -1267,6 +1267,45 @@ def _repeat2(input_iter: Iterator[TensorDict], **kwargs) -> Iterator[TensorDict]
     assert func(2) == 21
 
 
+def _repeat2(input_iter: Iterator[TensorDict], **kwargs) -> Iterator[TensorDict]:
+    for tdict in input_iter:
+        yield tdict
+        yield tdict
+
+
+def test_MultiProcPostprocessingDataset():
+    from test_HDFDataset import generate_hdf_from_dummy
+
+    num_hdfs = 20
+    num_seqs = 23
+    total_num_seqs = num_hdfs * num_seqs
+    total_num_seqs_pp = 2 * total_num_seqs
+    # Create a few HDF files such that we can easily verify the data later.
+    hdf_files = [generate_hdf_from_dummy() for _ in range(num_hdfs)]
```

Comment:
This is a bit problematic. You will have duplicates then in the seq tags. It might not really matter, but I wonder.

Why to use HDF at all here? Is this relevant for the test? Why not use just sth like this:
```python
{"class": "DummyDataset", "input_dim": 13, "output_dim": 7, "num_seqs": total_num_seqs, "seq_len": 17}
```

Reply:
Yeah, is fine. The test does not care about this 👍